response_operations_ui/controllers/uaa_controller.py

```diff
@@ -50,3 +50,64 @@ def sign_in(username, password):
     except (JSONDecodeError, ValueError):
         logger.exception("Error decoding JSON response")
         abort(500)
+
+
+def login_admin():
+    headers = {'Content-Type': 'application/x-www-form-urlencoded',
+               'Accept': 'application/json'}
+    payload = {'grant_type': 'client_credentials',
+               'response_type': 'token',
+               'token_format': 'opaque'}
+    try:
+        url = f"{app.config['UAA_SERVICE_URL']}/oauth/token"
+        response = requests.post(url, headers=headers, params=payload,
+                                 auth=(app.config['UAA_CLIENT_ID'], app.config['UAA_CLIENT_SECRET']))
+        resp_json = response.json()
+        return resp_json.get('access_token')
+    except HTTPError:
+        logger.exception(f'Failed to log into UAA', status_code=response.status_code)
+        abort(response.status_code)
+
+
+def get_user_by_email(email):
+    headers = {'Content-Type': 'application/json',
+               'Accept': 'application/json',
+               'Authorization': 'Bearer {}'.format(login_admin())}
+
+    url = f"{app.config['UAA_SERVICE_URL']}/Users?filter=email+eq+%22{email}%22"
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200 or response.json()['totalResults'] == 0:
+        return
+
+    return response.json()
+
+
+def get_first_name_by_email(email):
+    response = get_user_by_email(email)
+    if response is not None:
+        return response['resources'][0]['name']['givenName']
```

Comment:
Will all of these things be guaranteed to be there?  If givenName is empty, will the key still be there for it?
I only ask because debugging this becomes a pain as there are potentially 3 KeyErrors and 1 IndexError that could occur if the data isn't in the exact shape that you want

Reply:
They should be unless the UAA API changes